### PR TITLE
feat(daemon): allow pull from docker hub

### PR DESCRIPTION
**Pull request by deepankarm**
This PR fixes the following items. 
- Made `default` (cpu) the default build for all jinad containers. Added env vars for `DEVEL` builds, to be used with ci & cd. 
- Support only `.jinad` files (removed support for files like `random.jinad`)  
- Set `--store` to true by default. 

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200479679796904/1200528154584717) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by deepankarm** [comment]
FIxes #2740

**Review by jacobowitz — DISMISSED** [review]
lgtm :+1:

**Comment by jacobowitz** [comment]
rebased

**Review by jacobowitz — APPROVED** [review]
tests finally fixed

**Review by hanxiao — APPROVED** [review]

File: daemon/parser.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def mixin_daemon_parser(parser):
     gp.add_argument(
         '--store',
         action='store_true',
-        default=False,
+        default=True,
```
Comment (hanxiao) [inline daemon/parser.py:29]: using `store_true` action with `default=True` is basically making `store` as a `True` constant

Comment (jacobowitz) [reply daemon/parser.py:29]: yeah maybe should actually remove this option. Not sure we still need it.

**Pull request merged**